Assets/ApplicationLauncherManager.cs

```diff
@@ -0,0 +1,103 @@
+using System.Diagnostics;
+using System.IO;
+using System;
+using UnityEngine;
+
+public class ApplicationLauncherManager : MonoBehaviour
+{
+    private static ApplicationLauncherManager _instance;
+    public static ApplicationLauncherManager Instance
+    {
+        get
+        {
+            if (_instance == null)
+            {
+                var obj = new GameObject("ApplicationLauncherManager");
+                _instance = obj.AddComponent<ApplicationLauncherManager>();
+                DontDestroyOnLoad(obj);
+            }
+            return _instance;
+        }
+    }
+private void HideLauncherUI()
+{
+    UnityEngine.Debug.Log("Hiding launcher UI");
+}
+
+private void ShowLauncherUI()
+{
+    UnityEngine.Debug.Log("Showing launcher UI");
+}
+
+private void OnApplicationExited(object sender, EventArgs e)
+{
+    UnityEngine.Debug.Log("Application exited. Returning to launcher.");
+    ShowLauncherUI();
+}
+
+private Process currentProcess;
+
+    /// <summary>
+    /// Launches an external application or game from the given path.
+    /// </summary>
+    /// <param name="exePath">Absolute path to the EXE or shortcut</param>
+public void LaunchApplication(string appPath)
+    {
+    #if UNITY_STANDALONE_OSX
+        if (!Directory.Exists(appPath))
+        {
+            UnityEngine.Debug.LogError($"App not Found: {appPath}");
+            return;
+        }
+        try
+        {
+            ProcessStartInfo startInfo = new ProcessStartInfo
```

Comment:
Given the only difference between this and line 82 is the extra argument and FileName, I think it'd be better to wrap the ProcessStartInfo around the preprocessor checks and change the object that the rest of the script will use instead of having two separate blocks that end up running the same code.